core/ir.ml

```diff
@@ -71,10 +71,11 @@ and binding =
 and special =
   | Wrong      of Types.datatype
   | Database   of value
-  | Lens       of value * Lens.Sort.t
-  | LensDrop   of value * string * string * value * Lens.Sort.t
-  | LensSelect of value * Lens.Phrase.t * Lens.Sort.t
-  | LensJoin   of value * value * string list * Lens.Phrase.t * Lens.Phrase.t * Lens.Sort.t
+  | Lens       of value * Lens.Type.t
+  | LensDrop   of value * string * string * value * Lens.Type.t
+  | LensSelect of value * [`Static of Lens.Phrase.t | `Dynamic of value] * Lens.Type.t
+  | LensJoin   of value * value * string list * Lens.Phrase.t * Lens.Phrase.t * Lens.Type.t
```

Comment:
What is the `string list` referring to? I'd recommend either adding a comment, adding a type alias, or (better) making this long payload a record with descriptive names